todoman/cli.py

```diff
@@ -325,12 +337,22 @@ def list(ctx, lists, all, urgent, location, category, grep, sort, reverse):
                (not category or category in todo.categories) and
                (not pattern or (
                    pattern.search(todo.summary) or
-                   pattern.search(todo.description)
-                   ))
+                   pattern.search(todo.description))) and
+               (not due or
+                (todo.due and
+                 not todo.due.tzinfo and
+                 (todo.due <= max_naive)) or
+                (todo.due and
+                 todo.due.tzinfo and
+                 (todo.due <= max_aware)))
```

Comment:
IIRC, python does some magic optimizations for this sort of list comprehension, I'd have to see if they still apply here, and if a function has any drawbacks (I admit I'm a bit ignorant on this, and might be mistaken).